From 279e0b7a7e944269e93c992c9becdec35ed71401 Mon Sep 17 00:00:00 2001
Message-Id: <279e0b7a7e944269e93c992c9becdec35ed71401.1431284210.git.chang-joon.lee@intel.com>
In-Reply-To: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
References: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
From: Xiujun Geng <xiujun.geng@intel.com>
Date: Thu, 7 May 2015 18:15:54 +0800
Subject: [PATCH 67/68] SQUASHME! [VPG]: drm/i915: update dsparb2 and dsphowm
 to default value

dsparb2 and dsphowm need to be updated to default value othwerwise the
fifo allocation may be messyed up.

This patch should be squashed to:
    commit e4baa1d2ad36ef7d6af9998af238c51676290ed5
    Author: vandita kulkarni <vandita.kulkarni@intel.com>
    Date:   Sat Apr 18 00:12:16 2015 +0530

        REVERTME: [VPG]: drm/i915 : program dynamic dsparb and wm

Issue: IMINAN-33414
Change-Id: I001aee91f3b265deb8f530310ee138a331b74e6f
Signed-off-by: Xiujun Geng <xiujun.geng@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h      |    1 +
 drivers/gpu/drm/i915/intel_display.c |    2 ++
 2 files changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 6008968..62fd9a5 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -4206,6 +4206,7 @@ enum punit_power_well {
 #define	DSPARB_AEND_SHIFT	0
 #define	DSPARB2_VLV_DEFAULT	0x00111111
 #define	DSPARB3_VLV_DEFAULT	0x80008000
+#define DSPHOWM_VLV_DEFAULT     0x0
 #define DSPARB_50_25_25		0x8000
 #define DSPARB2_50_25_25	0x11
 #define DSPARB_PRI50_SA50	0xff00
diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 75df591..2c80741 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2619,6 +2619,8 @@ static void intel_find_plane_obj(struct intel_crtc *intel_crtc,
 void vlv_update_watermarks(struct drm_i915_private *dev_priv)
 {
 	I915_WRITE(DSPARB, DSPARB_VLV_DEFAULT);
+	I915_WRITE(DSPARB2, DSPARB2_VLV_DEFAULT);
+	I915_WRITE(DSPHOWM, DSPHOWM_VLV_DEFAULT);
 	I915_WRITE(DSPFW1,
 		   (DSPFW_SR_VAL << DSPFW_SR_SHIFT) |
 		   (DSPFW_CURSORB_VAL << DSPFW_CURSORB_SHIFT) |
-- 
1.7.9.5

